src/bin/rav1e-ch.rs

```diff
@@ -0,0 +1,574 @@
+        let mut buflen = [0u8; 8];
+
+        passfile
+          .read_exact(&mut buflen)
+          .map_err(|e| e.context("Unable to read the two-pass data file."))?;
+
+        let mut data = vec![0u8; u64::from_be_bytes(buflen) as usize];
+
+        passfile
+          .read_exact(&mut data)
+          .map_err(|e| e.context("Unable to read the two-pass data file."))?;
+
+        while send_rc.send(RcData::Frame(data.into_boxed_slice())).is_ok() {
+          passfile.read_exact(&mut buflen).map_err(|e| {
+            e.context("Unable to read the two-pass data file.")
+          })?;
+
+          if u64::from_be_bytes(buflen) == 0 {
+            break;
+          }
+
+          data = vec![0u8; u64::from_be_bytes(buflen) as usize];
+
+          passfile.read_exact(&mut data).map_err(|e| {
+            e.context("Unable to read the two-pass data file.")
+          })?;
+        }
+      }
+
+      Ok(())
+    });
+
+    // Receive Packets
+    let d = s.spawn(move |_| -> Result<(), CliError> {
+      for pkt in receive_packet.iter() {
+        output.write_frame(
+          pkt.input_frameno as u64,
+          pkt.data.as_ref(),
+          pkt.frame_type,
+        );
+        output.flush().unwrap();
+        if let (Some(ref mut y4m_enc_uw), Some(ref rec)) =
+          (y4m_enc.as_mut(), &pkt.rec)
+        {
+          write_y4m_frame(y4m_enc_uw, rec, y4m_details);
+        }
+        let summary = build_frame_summary(
+          pkt,
+          y4m_details.bit_depth,
+          y4m_details.chroma_sampling,
+          metrics_enabled,
+        );
+
+        if verbose != Verbose::Quiet {
+          progress.add_frame(summary.clone());
+          if verbose == Verbose::Verbose {
+            info!("{} - {}", summary, progress);
+          } else {
+            // Print a one-line progress indicator that overrides itself with every update
+            eprint!("\r{}                    ", progress);
+          };
+        }
+      }
+
+      if verbose != Verbose::Quiet {
+        if verbose == Verbose::Verbose {
+          // Clear out the temporary progress indicator
+          eprint!("\r");
+        }
+        progress.print_summary(verbose == Verbose::Verbose);
+      }
+
+      // receive_packet.result()
+      Ok(())
+    });
+
+    a.join().unwrap()?;
+    b.join().unwrap()?;
+    c.join().unwrap()?;
+    d.join().unwrap()?;
+
+    Ok(())
+  })
+  .unwrap()
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+  #[cfg(feature = "tracing")]
+  use rust_hawktracer::*;
+  init_logger();
+
+  #[cfg(feature = "tracing")]
+  let instance = HawktracerInstance::new();
+  #[cfg(feature = "tracing")]
+  let _listener = instance.create_listener(HawktracerListenerType::ToFile {
+    file_path: "trace.bin".into(),
+    buffer_size: 4096,
+  });
+
+  run().map_err(|e| {
+    error::print_error(&e);
+    Box::new(e) as Box<dyn std::error::Error>
+  })
+}
+
+fn init_logger() {
+  use std::str::FromStr;
```

Comment:
why having the use statement in the function and not on top?

Reply:
because it is used only in one place.